core/tests/Test.Microsoft.Identity.Core.UIAutomation/CoreMobileTestHelper.cs

```diff
@@ -0,0 +1,104 @@
+
+            if (user.IsFederated)
+            {
+                switch (user.FederationProvider)
+                {
+                    case FederationProvider.AdfsV3:
+                    case FederationProvider.AdfsV4:
+                        passwordInputID = UiTestConstants.AdfsV4WebPasswordID;
+                        signInButtonID = UiTestConstants.AdfsV4WebSubmitID;
+                        break;
+                    default:
+                        passwordInputID = UiTestConstants.WebPasswordID;
+                        signInButtonID = UiTestConstants.WebSubmitID;
+                        break;
+                }
+            }
+            else
+            {
+                passwordInputID = UiTestConstants.WebPasswordID;
+                signInButtonID = UiTestConstants.WebSubmitID;
+            }
+
+            //Acquire token flow
+            controller.Tap(UiTestConstants.AcquireTokenID);
+            //i0116 = UPN text field on AAD sign in endpoint
+            controller.EnterText(UiTestConstants.WebUPNInputID, 20, user.Upn, true);
+            //idSIButton9 = Sign in button
+            controller.Tap(UiTestConstants.WebSubmitID, true);
+            //i0118 = password text field
+            controller.EnterText(passwordInputID, ((LabUser)user).GetPassword(), true);
+            controller.Tap(signInButtonID, true);
+        }
+
+        public static void VerifyResult(ITestController controller)
+        {
+            RetryVerificationHelper(() => {
+                //Test results are put into a label that is checked for messages
+                var result = controller.GetText(UiTestConstants.TestResultID);
+                if (result.Contains(UiTestConstants.TestResultSuccsesfulMessage))
+                {
+                    return;
+                }
+                else if (result.Contains(UiTestConstants.TestResultFailureMessage))
+                {
+                    throw new ResultVerificationFailureException(VerificationError.ResultIndicatesFailure);
+                }
+                else
+                {
+                    throw new ResultVerificationFailureException(VerificationError.ResultNotFound);
+                }
+            });
+
+        }
+
+        private static void RetryVerificationHelper(Action verification)
+        {
+            //There may be a delay in the amount of time it takes for an authentication request to complete.
+            //Thus this method will check the result once a second for 20 seconds.
+            var attempts = 0;
+            do
+            {
+                try
+                {
+                    attempts++;
+                    verification();
+                    break;
+                }
+                catch (ResultVerificationFailureException ex)
+                {
+                    if (attempts == UiTestConstants.maximumResultCheckRetryAttempts)
+                        throw new Exception("Could not Verify test result", ex);
```

Comment:
>throw new Exception("Could not Verify test result", ex); [](start = 24, length = 56)

before you moved it you mentioned this could be an assert instead... did you change your mind?

Reply:
Yes, I realized that there wouldn't be much value in using an exception rather than Assert.Fail().